### PR TITLE
Feat/processor filter

### PR DESCRIPTION
**Pull request by gitsimon**

### CODE REVIEW

File: pallets/marketplace/Cargo.toml [hunk 1]
```diff
@@ -28,9 +28,17 @@ sp-std = { git = "https://github.com/paritytech/substrate", default-features = f
 sp-runtime = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.41" }
 sp-io = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.41" }
 sp-core = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.41" }
+jsonrpsee = { version = "0.16.2", features = ["client-core", "server", "macros"], optional = true }
+sp-blockchain = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.36", optional = true }
```
Comment (RomarQ) [inline pallets/marketplace/Cargo.toml:32]:
```suggestion
sp-blockchain = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.41", optional = true }
```

File: pallets/marketplace/src/types.rs [hunk 2]
```diff
@@ -235,3 +270,37 @@ impl<T: Config> MarketplaceHooks<T> for () {
...
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
+#[derive(RuntimeDebug, codec::Encode, codec::Decode, PartialEq, Eq)]
+pub enum RuntimeApiError {
+    /// Error when filtering matching sources failed.
+    #[cfg_attr(feature = "std", error("Filtering matching sources failed."))]
+    FilterMatchingSources,
+}
+
+impl RuntimeApiError {
+    #![allow(unused_variables)]
```
Comment (RomarQ) [inline pallets/marketplace/src/types.rs:284]: There are no unused vars here.
```suggestion
```

File: pallets/processor-manager/src/lib.rs [hunk 3]
```diff
@@ -129,10 +132,16 @@ pub mod pallet {
     pub(super) type ManagerCounter<T: Config> = StorageMap<_, Blake2_128, T::AccountId, T::Counter>;
 
     #[pallet::storage]
-    #[pallet::getter(fn processor_last_seen)]
     pub(super) type ProcessorHeartbeat<T: Config> = StorageMap<_, Blake2_128, T::AccountId, u128>;
 
+    #[pallet::storage]
+    #[pallet::getter(fn processor_last_seen)]
+    pub(super) type ProcessorLastSeen<T: Config> = StorageMap<_, Identity, T::AccountId, u128>;
```
Comment (RomarQ) [inline pallets/processor-manager/src/lib.rs:139]: I do not get why we need to rename this storage map. Their names are equivalent in this context and there is no point in messing the storage with a migration.

**Review by RomarQ — CHANGES_REQUESTED** [review]

File: pallets/processor-manager/src/lib.rs (hunk 3 above)
Comment (gitsimon) [reply pallets/processor-manager/src/lib.rs]: the idea was to iterate over all processors and filter the once to display for selection, but turns out iteration is not possible.
For now I removed the migration to a reversible key.

File: pallets/marketplace/Cargo.toml (hunk 1 above)
Comment (gitsimon) [reply pallets/marketplace/Cargo.toml]: done

File: pallets/marketplace/src/types.rs (hunk 2 above)
Comment (gitsimon) [reply pallets/marketplace/src/types.rs]: removed

File: pallets/marketplace/src/rpc.rs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    AccountId: MaybeSerializeDeserialize,
+>
+{
+    /// Filters the given `sources` by those recently seen and matching partially specified `registration`.
+    #[method(name = "filterMatchingSources")]
+    fn filter_matching_sources(
+        &self,
+        registration: PartialJobRegistration<R, AccountId>,
+        sources: Vec<AccountId>,
+        // consumer: Option<MultiOrigin<AccountId>>,
```
Comment (RomarQ) [inline pallets/marketplace/src/rpc.rs:38]:
```suggestion
```

File: pallets/marketplace/src/types.rs [hunk 5]
```diff
@@ -235,3 +270,36 @@ impl<T: Config> MarketplaceHooks<T> for () {
...
+            e,
+        );
+        self
+    }
+
+    /// Consume given error `e` with `self` and generate a native log entry with error details.
+    pub fn log_debug(self, e: impl Debug) -> Self {
+        log::debug!(
+            target: "runtime::mmr",
+            "[{:?}] MMR error: {:?}",
```
Comment (RomarQ) [inline pallets/marketplace/src/types.rs:299]: This change is not related to `mmr`.

**Review by RomarQ — CHANGES_REQUESTED** [review]

**Review by godenzim — APPROVED** [review]

File: pallets/marketplace/src/rpc.rs (hunk 4 above)
Comment (gitsimon) [reply pallets/marketplace/src/rpc.rs]: consumer parameter is now included

File: pallets/marketplace/src/types.rs (hunk 5 above)
Comment (gitsimon) [reply pallets/marketplace/src/types.rs]: improved

**Review by godenzim — APPROVED** [review]

**Review by RomarQ — APPROVED** [review]
